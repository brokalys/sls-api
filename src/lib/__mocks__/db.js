export default {
  connect: jest.fn(),
  getClient: jest.fn(() => ({
    escape: jest.fn(),
  })),
  query: jest.fn(() => []),
  end: jest.fn(),
};
