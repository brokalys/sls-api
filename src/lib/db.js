import serverlessMysql from 'serverless-mysql';

const mysql = serverlessMysql({
  config: {
    host: process.env.DB_HOST,
    database: process.env.DB_DATABASE,
    user: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
  },
  maxRetries: 5,
});

if (process.env.NODE_ENV !== 'test') {
  mysql.connect();
}

export default mysql;
