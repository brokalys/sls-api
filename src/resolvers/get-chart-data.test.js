import db from '../lib/db';
import getChartData from './get-chart-data';

jest.mock('../lib/cache');
jest.mock('../lib/db');

db.query.mockImplementation(() => [
  {
    price: 110000.0,
    lat: 56.9366684,
    lng: 24.0794235,
    price_per_sqm: 100,
    area: 120,
    area_measurement: 'm2',
    published_at: '2018-12-01',
  },
  {
    price: 170000.0,
    lat: 56.9366684,
    lng: 24.0794235,
    price_per_sqm: 0,
    area: 1,
    area_measurement: 'ha',
    published_at: '2018-12-01',
  },
  {
    price: 70000.0,
    lat: 56.9366684,
    lng: 24.0794235,
    price_per_sqm: 0,
    area: 120,
    area_measurement: 'm2',
    published_at: '2018-11-01',
  },
]);

describe('getChartData', () => {
  describe('response', () => {
    test('returns all chart data with only category', async () => {
      const output = await getChartData(
        {},
        { category: 'APARTMENT', date: '2018-01-01' },
      );

      expect(output).toMatchSnapshot();
    });

    test('returns all chart data with category + type', async () => {
      const output = await getChartData(
        {},
        {
          category: 'APARTMENT',
          type: 'SELL',
          date: '2018-01-01',
        },
      );

      expect(output).toMatchSnapshot();
    });
  });
});
