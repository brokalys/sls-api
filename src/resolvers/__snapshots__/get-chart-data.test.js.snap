// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getChartData response returns all chart data with category + type 1`] = `
Array [
  Object {
    "count": 0,
    "date": "2018-01-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-02-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-03-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-04-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-05-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-06-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-07-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-08-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-09-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-10-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 1,
    "date": "2018-11-01",
    "price_per_sqm": "583.33",
  },
  Object {
    "count": 2,
    "date": "2018-12-01",
    "price_per_sqm": "50.00",
  },
]
`;

exports[`getChartData response returns all chart data with only category 1`] = `
Array [
  Object {
    "count": 0,
    "date": "2018-01-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-02-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-03-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-04-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-05-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-06-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-07-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-08-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-09-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 0,
    "date": "2018-10-01",
    "price_per_sqm": "0.00",
  },
  Object {
    "count": 1,
    "date": "2018-11-01",
    "price_per_sqm": "583.33",
  },
  Object {
    "count": 2,
    "date": "2018-12-01",
    "price_per_sqm": "50.00",
  },
]
`;
